Arcitecture_Components/app/src/main/java/com/example/arcitecture_components/MainActivity.java

```diff
@@ -0,0 +1,275 @@
+                recyclerView.setAdapter(recycler);*/
+               /* handleCountryList(countries);
+                }
+        });
+            countryListViewModel.getCountryResponseLive().observe(this, new Observer<CountryListViewModel.RequestStatus>() {
+            @Override
+            public void onChanged(CountryListViewModel.RequestStatus requestStatus) {
+                dialog = new ProgressDialog(MainActivity.this);
+                switch (requestStatus)
+                {
+                    case IN_PROGRESS:
+
+                        dialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
+                        dialog.setTitle("Fetching games");
+                        dialog.setMessage("Please wait...");
+                        dialog.setIndeterminate(true);
+                        dialog.setCanceledOnTouchOutside(false);
+                        dialog.show();
+                        break;
+                    case SUCCEEDED:
+                        dialog.dismiss();
+                        break;
+                    case FAILED:
+                    //ad = new AlertDialog.Builder(MainActivity.this).setTitle("Request Failed").setMessage("Fetching country request failed");
+                    //ad.show();
+
+
+                }
+            }
+        });*/
+    }
+    private void setUpLiveData() {
+        countryListViewModel.getCountryListLive().observe(this, new Observer<List<Countries>>() {
+            @Override
+            public void onChanged(List<Countries> countriesList) {
+                handleCountryList(countriesList);
+            }
+        });
+        countryListViewModel.getCountryResponseLive().observe(this, new Observer<CountryListViewModel.RequestStatus>() {
+            @Override
+            public void onChanged(CountryListViewModel.RequestStatus requestStatus) {
+                handleRequestStatus(requestStatus);
+            }
+        });
+        countryListViewModel.getcolor().observe(this, new Observer<Boolean>() {
+            @Override
+            public void onChanged(Boolean aBoolean) {
+                //if(aBoolean)
+                //{
+                    ConstraintLayout constraintLayout = findViewById(R.id.Container);
+                    constraintLayout.setBackgroundResource(R.color.purple_200);
+                //}
+            }
+        });
+
+    }
+    private void handleCountryList(List<Countries> countries) {
+        RecyclerAdapter recycler =new RecyclerAdapter(countries);
+        RecyclerView recyclerView =findViewById(R.id.Recycle);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(new LinearLayoutManager(MainActivity.this));
+        recyclerView.setAdapter(recycler);
+        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(this, DividerItemDecoration.VERTICAL);
+        recyclerView.addItemDecoration(dividerItemDecoration);
```

Comment:
Avoid initializing the adapter inside livedata callback. Move to onCreateView. Only update the adapter data here using  a method like `updateQuizList`.